Mathlib/LinearAlgebra/PerfectPairing.lean

```diff
@@ -187,12 +192,17 @@ theorem finrank_eq [Module.Finite R M] [Module.Free R M] :
     finrank R M = finrank R N :=
   ((Module.Free.chooseBasis R M).toDualEquiv.trans p.toDualRight.symm).finrank_eq
 
-private lemma restrict_aux
-    {M' N' : Type*} [AddCommGroup M'] [Module R M'] [AddCommGroup N'] [Module R N']
-    (i : M' →ₗ[R] M) (j : N' →ₗ[R] N)
-    (hM : IsCompl (LinearMap.range i) ((LinearMap.range j).dualAnnihilator.map p.toDualLeft.symm))
-    (hN : IsCompl (LinearMap.range j) ((LinearMap.range i).dualAnnihilator.map p.toDualRight.symm))
-    (hi : Injective i) (hj : Injective j) :
+section Restrict
+
+variable {M' N' : Type*} [AddCommGroup M'] [Module R M'] [AddCommGroup N'] [Module R N']
+  (i : M' →ₗ[R] M) (j : N' →ₗ[R] N)
+  (hM : IsCompl (LinearMap.range i) ((LinearMap.range j).dualAnnihilator.map p.toDualLeft.symm))
+  (hN : IsCompl (LinearMap.range j) ((LinearMap.range i).dualAnnihilator.map p.toDualRight.symm))
```

Comment:
I have been toying with introducing:
```lean
structure IsAnnCompl (U : Submodule R M) (V : Submodule R N) : Prop where
  left : IsCompl U (V.dualAnnihilator.map p.toDualLeft.symm)
  right : IsCompl V (U.dualAnnihilator.map p.toDualRight.symm)
```
to express this since it has now come up a few times.

Reply:
Yeah, that seems reasonable to me.

Reply:
I proposed the name `PerfectPairing.IsPerfectCompl`